sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py

```diff
@@ -181,11 +200,21 @@ async def get_schema_id(
         except AttributeError:
             pass
 
-        return await self._generated_client.schema.query_id_by_content(
-            group_name=schema_group,
-            schema_name=schema_name,
-            schema_content=schema_content,
-            x_schema_type=serialization_type,
-            cls=_parse_response_schema_id,
-            **kwargs
-        )
+        try:
+            return self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ]
+        except KeyError:
+            schema_properties = await self._generated_client.schema.query_id_by_content(
+                group_name=schema_group,
+                schema_name=schema_name,
+                schema_content=schema_content,
+                x_schema_type=serialization_type,
+                cls=_parse_response_schema_id,
+                **kwargs
+            )
+            self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ] = schema_properties
+            self._id_to_schema[schema_properties.schema_id] = Schema(schema_content, schema_properties)
```

Comment:
it is possible that schema_properties.schema_id is already in the `self._id_to_schame` such that we're overwriting the item here?.

e.g. if a user calls `get_schema` first, then call `get_schema_id`

Reply:
same response as above